File: gamepack/Decompiled/Bounds.java
public class Bounds {
   public int height;
   public int x;
   public int y;
   public int width;

   public Bounds(int var1, int var2) {
      this(0, 0, var1, var2);
   }

   public Bounds(int var1, int var2, int var3, int var4) {
      this.setLocation(var1, var2);
      this.setSize(var3, var4);
   }

   public void setLocation(int var1, int var2) {
      this.x = var1 * 775648579;
      this.y = -1859334611 * var2;
   }

   public void setSize(int var1, int var2) {
      this.width = -1469434281 * var1;
      this.height = 149296003 * var2;
   }

   public boolean aw(int var1, int var2) {
      return var1 >= -844906645 * this.x && var1 < this.width * 944313703 + this.x * -844906645 && var2 >= this.y * 1864020389 && var2 < this.y * 1864020389 + this.height * -1376251093;
   }

   public String toString() {
      return null;
   }

   public void ac(Bounds var1, Bounds var2) {
      this.au(var1, var2);
      this.ab(var1, var2);
   }

   void au(Bounds var1, Bounds var2) {
      var2.x = 1 * this.x;
      var2.width = this.width * 1;
      if (this.x * -844906645 < var1.x * -844906645) {
         var2.width -= -413884579 * var1.x - this.x * -413884579;
         var2.x = 1 * var1.x;
      }

      if (var2.aq() > var1.aq()) {
         var2.width -= (var2.aq() - var1.aq()) * -1469434281;
      }

      if (var2.width * 944313703 < 0) {
         var2.width = 0;
      }

   }

   void ab(Bounds var1, Bounds var2) {
      var2.y = this.y * 1;
      var2.height = this.height * 1;
      if (this.y * 1864020389 < var1.y * 1864020389) {
         var2.height -= var1.y * -1743378833 - -1743378833 * this.y;
         var2.y = var1.y * 1;
      }

      if (var2.al() > var1.al()) {
         var2.height -= (var2.al() - var1.al()) * 149296003;
      }

      if (var2.height * -1376251093 < 0) {
         var2.height = 0;
      }

   }

   int aq() {
      return this.width * 944313703 + this.x * -844906645;
   }

   int al() {
      return -1376251093 * this.height + this.y * 1864020389;
   }
}

File: find-names/src/main/java/mapper/identifiers/classes/Bounds.kt
package mapper.identifiers.classes

import mapper.abstractclasses.IdentityMapper
import mapper.abstractclasses.OrderMapper
import mapper.annotations.DependsOn
import mapper.annotations.MethodParameters
import mapper.predicateutilities.and
import mapper.predicateutilities.predicateOf
import mapper.predicateutilities.type
import mapper.wrappers.ClassWrapper
import mapper.wrappers.InstructionMapper
import mapper.wrappers.MethodWrapper
import org.objectweb.asm.Opcodes
import org.objectweb.asm.Type
import java.lang.reflect.Modifier

class Bounds : IdentityMapper.Class() {
    override val predicate = predicateOf<ClassWrapper> { it.superType == Any::class.java.type }
            .and { it.interfaces.isEmpty() }
            .and { it.instanceFields.size == 4 }
            .and { it.instanceFields.all { it.type == Type.INT_TYPE } }
            .and { it.instanceMethods.size == 9 }
            .and { !Modifier.isAbstract(it.access) }

    @MethodParameters("x", "y")
    class setLocation : OrderMapper.InConstructor.Method(Bounds::class, 0, 2) {
        override val constructorPredicate = predicateOf<MethodWrapper> { it.arguments.size == 4 }
        override val predicate = predicateOf<InstructionMapper> { it.opcode == Opcodes.INVOKEVIRTUAL }
    }

    @MethodParameters("width", "height")
    class setSize : OrderMapper.InConstructor.Method(Bounds::class, 1, 2) {
        override val constructorPredicate = predicateOf<MethodWrapper> { it.arguments.size == 4 }
        override val predicate = predicateOf<InstructionMapper> { it.opcode == Opcodes.INVOKEVIRTUAL }
    }

    @DependsOn(setLocation::class)
    class x : OrderMapper.InMethod.Field(setLocation::class, 0, 2) {
        override val predicate = predicateOf<InstructionMapper> { it.opcode == Opcodes.PUTFIELD }
    }

    @DependsOn(setLocation::class)
    class y : OrderMapper.InMethod.Field(setLocation::class, 1, 2) {
        override val predicate = predicateOf<InstructionMapper> { it.opcode == Opcodes.PUTFIELD }
    }

    @DependsOn(setSize::class)
    class width : OrderMapper.InMethod.Field(setSize::class, 0, 2) {
        override val predicate = predicateOf<InstructionMapper> { it.opcode == Opcodes.PUTFIELD }
    }

    @DependsOn(setSize::class)
    class height : OrderMapper.InMethod.Field(setSize::class, 1, 2) {
        override val predicate = predicateOf<InstructionMapper> { it.opcode == Opcodes.PUTFIELD }
    }
}